File: recipes-for-diabetics/src/app/pojo/ingredient.ts
import { Nutrition } from "./nutrition";
import { Quantity } from "./quantity";

export class Ingredient {
    name:string;
    url:string;
    nutrition:Nutrition;
    quantity:Quantity
    constructor(name:string, url:string, nutrition:Nutrition, quantity:Quantity) {
        this.name = name;
        this.url = url;
        this.nutrition = nutrition;
        this.quantity = quantity;
    }
}

File: identity-server/src/main/java/com/avasthi/varahamihir/identityserver/entities/Ingredient.java
package com.avasthi.varahamihir.identityserver.entities;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Objects;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public final class Ingredient extends AbstractBase {
    private String name;
    private String description;
    private String brand;
    private String url;
    private Quantity quantity;
}
